### PR TITLE
[#1791] Switch to package-lock.json and npm ci

### LINKED ISSUE #1791: Continuous Integration: Switch to package-lock.json and npm ci
**What feature(s) would you like to see in RepoSense?**

As mentioned in https://github.com/reposense/RepoSense/pull/1785#issue-1262024127, we should consider switching to using [`package-lock.json`](https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json) and [`npm ci`](https://docs.npmjs.com/cli/v8/commands/npm-ci) to ensure that dependency versions are fixed when running CI. This allows for CI builds to be repeatable and consistent. 

**Is the feature request related to a problem?**

#1784 demonstrated how changes to upstream technologies and dependencies can break CI. Although #1784 is more likely to be due to Chrome updating to version 102, there is also a possibility that updating a dependency during a CI run might cause a breaking change. 

**If possible, describe the solution**

Right now, only `frontend/cypress` has the `package-lock.json` tracked by git. There is a need to re-add the `package-lock.json` in the `frontend` directory to git tracking.

At the same time, the `installFrontend` task in `build.gradle` should be modified so that `npm ci` is run.

**Additional context**

By switching to `npm ci`, updating dependencies would need to be done manually.

### PR DESCRIPTION
**Pull request by MarcusTXK**
Fixes #1791 
## Proposed commit message

```
Issue 1784 demonstrates how changes to upstream technologies and
dependencies can break CI.  Although it was more likely to be due to
Chrome updating to version 102, there is also a possibility that
updating a dependency during a CI run might cause a breaking change. 

Let's consider switching to using package-lock.json and npm ci to ensure
that dependency versions are fixed when running CI. This allows for CI
builds to be repeatable and consistent.
```

## Other information

Currently, there are significant overlaps between the 2 `.gitignore` files in the root folder as well as the frontend. I suggest we consolidate it into a single .gitignore file for easier maintainability. This is done in PR #1869.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -59,7 +59,7 @@ wrapper {
 
 def installFrontend = tasks.register('installFrontend', com.liferay.gradle.plugins.node.tasks.ExecutePackageManagerTask) {
     workingDir 'frontend/'
-    args = ['install']
+    args = ['ci']
```
Comment (yhtMinceraft1010X) [inline build.gradle:62]: There's an `installCypress` task at line 188 that also needs to have the args changed from `['install']` to `['ci']`

**Review by yhtMinceraft1010X — CHANGES_REQUESTED** [review]
Feel free to consolidate the .gitignore files into 1.

File: build.gradle (hunk 1 above)
Comment (yhtMinceraft1010X) [reply build.gradle:62]: Please check your commit message character count. The longest line has 101 characters.

Comment (MarcusTXK) [reply build.gradle:62]: Apologies for missing that, I will update that as well.

**Review by yhtMinceraft1010X — APPROVED** [review]

**Review by yhtMinceraft1010X — CHANGES_REQUESTED** [review]
I apologise for this sudden backtracking of stance, but now that I think of it, the .gitignore consolidation should be reversed for now and moved into a new PR.

This is to keep the scope of changes small enough that any future reversals of either change won't affect the other.

**Review by yhtMinceraft1010X — APPROVED** [review]
LGTM

**Review by chan-j-d — APPROVED** [review]
LGTM!

**Comment by yhtMinceraft1010X** [comment]
Note to anyone who's merging this, please merge #1869 first

**Comment by dcshzj** [comment]
I have merged #1869, do resolve the file conflicts, thanks!

**Comment by dcshzj** [comment]
Sorry @MarcusTXK just for a sanity check, which Node.js version did you use to generate the package-lock.json file? Our docs suggests using version 14.19.0.

**Comment by MarcusTXK** [comment]
> Sorry @MarcusTXK just for a sanity check, which Node.js version did you use to generate the package-lock.json file? Our docs suggests using version 14.19.0.

I was using v16.13.2, but I have swapped my version over to v14.19.0, deleted and regenerated the `package-lock.json` for the frontend to be safe. Thanks for checking.

**Pull request merged**